examples/shader/overriding_material_shader.rs

```diff
@@ -0,0 +1,148 @@
+use bevy::{
+    prelude::*,
+    reflect::TypeUuid,
+    render::{
+        pipeline::{PipelineDescriptor, RenderPipeline},
+        shader::{ShaderStage},
+        texture::{Extent3d,TextureDimension,TextureFormat},
+    },
+};
+fn main() {
+    App::build()
+        .add_plugins(DefaultPlugins)
+        .add_startup_system(setup.system())
+        .run();
+}
+//This is the handle for our custom pipeline, we have to create a new pipeline for our custom sprite rendering, or they won't get rendered
+//This is a pre-made handle so it will be easier to use
+pub const CUSTOM_SPRITE_PIPELINE_HANDLE: HandleUntyped =
+    HandleUntyped::weak_from_u64(PipelineDescriptor::TYPE_UUID, 2785347850338765446);
+
+//This bundle exists to make spawning shader-overriden sprites easier
+#[derive(Bundle)]
+struct CustomSpriteBundle {
+    #[bundle]
+    sprite: SpriteBundle,
+}
+impl Default for CustomSpriteBundle{
+    fn default() -> Self {
+        Self{
+            sprite: SpriteBundle {
+                //We make sure the sprite is rendered in our custom pipeline
+                render_pipelines: RenderPipelines::from_pipelines(vec![RenderPipeline::new(
+                    CUSTOM_SPRITE_PIPELINE_HANDLE.typed(),
+                )]),
+                ..Default::default()
+            },
+        }
+    }
+}
+impl CustomSpriteBundle {
+    fn new(material_handle: Handle<ColorMaterial>) -> Self {
+        let mut default = Self::default();
+        default.sprite.material = material_handle;
+        return default;
+    }
+}
+//We're copying everything from the original fragment shader, but changing stuff in the main() function of the shader
+//If the shader isn't working, probably something changed in the original vertex and fragment shaders in the version you are using
+//In your own shader, you could write it in a 'your-shader-name'.frag file and do like what is done in bevy_sprite/src/render/mod.rs (it's prettier, and you can get syntax highlighting for glsl, but i don't want to clutter the examples folder) 
```

Comment:
I would prefer this to be along the lines of:
"For your own shaders, consider writing them in their own files (see e.g. bevy_sprite/src/render/)"